rfcs/text/0000-extend-file-pe.md

```diff
@@ -0,0 +1,1076 @@
+# 0000: Extend the PE subfield of the File fieldset
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+The Portable Executable (PE) sub-field, of the `file` top-level fieldset, can be updated to include more file attributes to aid in file analysis. This additional document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create 25 additional sub-fields within the `file.pe` fieldset.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.pe.authentihash | keyword | Authentihash of the PE file. |
+| file.pe.compile_timestamp | date | Compile timestamp of the PE file. |
+| file.pe.compiler_product_versions | keyword | Version of the compiler. |
+| file.pe.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.pe.entry_point | long | Entry point of the PE file. |
+| file.pe.exports | keyword | List of symbols exported by PE |
+| file.pe.debug | flattened | Debug information, if present |
+| file.pe.import_list | flattened | List of all imported functions |
+| file.pe.sections | flattened | Data about sections of compiled binary PE |
```

Comment:
Similiar to the imports, sections are common to all executable/library types. Moving this data out of a flattened field gives us more usefulness of it.

Additional discussion: https://github.com/elastic/beats/pull/21815#discussion_r515410709
LIEF Abstract definition for comparision: https://lief.quarkslab.com/doc/latest/api/python/abstract.html#section

Reply:
Is this how you'd like to present this; or do you want to extend the values into nested fields? I can adjust that table (which may be a bit stale).
```
    - name: sections
      level: extended
      description: >
        Data about sections of compiled binary PE
      type: keyword
      example: { "chi2" : 3027194, "virtual_address": 8192, "entropy": 6.24, "flags": "rx", "name": ".text", "raw_size": 198144 }
```

Reply:
wondering if we could get `sections` specified as a `nested` field in this PR and then maybe that could lead us to a discussion of the various overlaps in subfields to drive a conversation around extraction into a known/documented common field set.

Reply:
Is this the way you'd prefer to represent this?
```
    - name: sections
      level: extended
      type: object
      description: Data about sections of compiled binary PE
      fields:
        - name: chi2
           level: extended
           type: keyword
           description: description
           example: 3027194
...
```

Reply:
Did this as `nested`, just like `debug`.